extension/js/add_input_icon.js

```diff
@@ -1,33 +1,98 @@
-(function () {
-  for (const emailInput of document.querySelectorAll("input[type='email']")) {
-    const divEl = document.createElement("div");
-    divEl.classList.add("relay-icon");
-
-    const buttonEl = document.createElement("button");
-    buttonEl.id = "relay-button";
-    buttonEl.classList.add("relay-button");
-    buttonEl.type = "button";
-
-    const imgEl = document.createElement("img");
-    imgEl.src = browser.runtime.getURL("icons/placeholder-logo-beta.png");
-    imgEl.classList.add("relay-icon-img");
-    buttonEl.appendChild(imgEl);
-
-    buttonEl.addEventListener("click", async (e) => {
-      const newRelayAddressResponse = await browser.runtime.sendMessage({
-          method: "makeRelayAddress",
-        });
-
-        if (newRelayAddressResponse.status === 402) {
-          emailInput.value = "Too many email aliases.";
-          return;
-        }
-
-        navigator.clipboard.writeText(newRelayAddressResponse);
-        emailInput.value = newRelayAddressResponse;
+function addRelayIconToInput(emailInput) {
+  // remember the input's original parent element;
+  const emailInputOriginalParentEl = emailInput.parentElement;
+
+  // create new wrapping element;
+  const emailInputWrapper = document.createElement("div");
+  emailInputWrapper.classList.add("relay-email-input-wrapper");
+
+  const inputClone = emailInput.cloneNode();
+  // add padding to the cloned input so that input text
+  // is not covered up by the Relay icon
+  inputClone.style.paddingRight = "50px !important;"
+
+
+  emailInputWrapper.appendChild(inputClone);
+
+  const inputHeight = emailInput.clientHeight;
+
+  const divEl = document.createElement("div");
+  divEl.classList.add("relay-icon");
+  divEl.style.height = inputHeight+"px";
```

Comment:
It didn't break on Monitor but I've also been wondering about various CSP conflict possibilities. It's possible that because, at this point in the script, we haven't actually appended this element to anything, it's not in violation with the website's CSP.

Reply:
oh that's true! it may be that the add-on script appends style attributes AFTER the CSP has enforced?